net/blockwise/blockwise.go

```diff
@@ -14,34 +14,37 @@ import (
 	"github.com/patrickmn/go-cache"
 )
 
+// Block Opion value is represented: https://tools.ietf.org/html/rfc7959#section-2.2
+//  0
+//  0 1 2 3 4 5 6 7
+// +-+-+-+-+-+-+-+-+
+// |  NUM  |M| SZX |
+// +-+-+-+-+-+-+-+-+
+//  0                   1
+//  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5
+// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+// |          NUM          |M| SZX |
+// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//  0                   1                   2
+//  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3
+// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+// |                   NUM                 |M| SZX |
+// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
 const (
-	maxBlockNumber = int(1048575)
+	// max block size is 3bytes: https://tools.ietf.org/html/rfc7959#section-2.1
+	maxBlockValue = 0xffffff
+	// maxBlockNumber is 20bits (NUM)
+	maxBlockNumber = 0xffff7
```

Comment:
maxBlockNumber was `1048575` == `0xFFFFF` not `0xffff7`

Reply:
there was a bug  - the max block number can have 20bits (0xffff7)